app/operations/ingredients/process.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+module Ingredients
+  # Splits ingredients into Ingredient data type. Combines like ingredients
+  # into Ingredient_to_serving data type. Returns all ingredients
+  # in an Ingredient_with_serving data type
+  class Process < ApplicationOperation
+    # @param ingredients [Array<Hash>] List of hashes with a single `:ingredient` string key.
+    # @param serving_size [Integer] Number of servings to scale the ingredient quantities to.
+    # @return [Dry::Monads::Result::Success<Ingredient_return>, Dry::Monads::Result::Failure]
+    # @example
+    #   Ingredients::Process.call(
+    #     ingredients: [
+    #       { ingredient: '1/2 pound of cabbage' },
+    #       { ingredient: '2 cups of butter' }
+    #     ],
+    #     serving_size: 2
+    #   )
+
+    def call(ingredients:, serving_size:)
+      ingredients = step split(ingredients)
+      step combine_ingredients(ingredients, serving_size)
+    end
+
+    private
+
+    def combine_ingredients(ingredients, serving_size)
+      totals = ingredients.each_with_object(Hash.new(0)) do |item, result|
+        result[item.name] += Measurements.to_tsp(item.amount, item.unit)
```

Comment:
I wonder if this would read better if `.to_tsp` just converted a unit to tsp, then you do the math itself, eg:

```suggestion
result[item.name] += Measurements.to_tsp(item.unit) * item.amount
```

Reply:
Yeah that makes sense.